tools/tools.mk

```diff
@@ -0,0 +1,107 @@
+# protoc and source is used to generate code from a protobuf file and handle imports
+PROTOC_VERSION=3.14.0
+PROTOC_ZIP=protoc-$(PROTOC_VERSION)-linux-x86_64.zip
+ifeq ($(GOHOSTOS)_$(GOHOSTARCH),darwin_amd64)
+    PROTOC_ZIP=protoc-$(PROTOC_VERSION)-osx-x86_64.zip
+endif
+PROTOC_RELEASES_URI=https://github.com/protocolbuffers/protobuf/releases/download
+PROTOC_DOWNLOAD=$(PROTOC_RELEASES_URI)/v$(PROTOC_VERSION)/$(PROTOC_ZIP)
+
+tools/protoc = $(TOOLS_BIN)/protoc/${PROTOC_VERSION}/protoc
+$(tools/protoc):
+	@printf "$(OK_COLOR)==> Installing tools/protoc$(NO_COLOR)\n"
+	mkdir -p "$(@D)"
+	curl --fail --output "$@.zip" -L "$(PROTOC_DOWNLOAD)"
+	go run tools/unzip.go "$@.zip" "$@" "bin/protoc"
+	rm -f "$@.zip"
+
+PROTOBUF_GZIP=protobuf-all-$(PROTOC_VERSION).tar.gz
+PROTOBUF_DOWNLOAD=$(PROTOC_RELEASES_URI)/v$(PROTOC_VERSION)/$(PROTOBUF_GZIP)
+
+tools/protobuf-src = $(TOOLS_VENDOR)/protobuf/${PROTOC_VERSION}/protobuf.tar.gz
+$(tools/protobuf-src): $(tools/protoc)
+	@printf "$(OK_COLOR)==> Installing tools/protobuf-src$(NO_COLOR)\n"
+	mkdir -p "$(@D)"
+	rm -rf $(TOOLS_VENDOR)/protobuf/current
+	curl --fail --output "$@" -L "$(PROTOBUF_DOWNLOAD)"
+	go run tools/untargzip.go "$@" "$(@D)"
+	mv $(TOOLS_VENDOR)/protobuf/${PROTOC_VERSION}/protobuf-${PROTOC_VERSION} $(TOOLS_VENDOR)/protobuf/current
```

Comment:
Should we clean up after the move?

Reply:
Makefile task refers to this file - it is marker of task completeness. We can remove it and use dummy file, but it is not big deal IMO.